crypto/src/openssl.rs

```diff
@@ -177,6 +194,28 @@ impl Crypto for OpensslCrypto {
         Ok(cdi)
     }
 
+    fn get_exported_cdi_handle(
+        &mut self,
+        cdi: &Self::Cdi,
+    ) -> Result<ExportedCdiHandle, CryptoError> {
+        for (stored_cdi, stored_handle) in self.export_cdi_slots.iter() {
+            if stored_cdi == cdi {
+                return Ok(stored_handle.to_owned());
+            }
+        }
+
+        if self.export_cdi_slots.len() >= MAX_CDI_HANDLES {
+            return Err(CryptoError::ExportedCdiHandleLimitExceeded);
+        }
+
+        let mut exported_cdi_handle = [0; MAX_EXPORTED_CDI_SIZE];
+        self.rand_bytes(&mut exported_cdi_handle)?;
+        self.export_cdi_slots
```

Comment:
Another option would be to generate a handle when we create the exported CDI and then just lookup the handle in this function. I don't think it matters much either way, but it's maybe a bit more clear to make the binding always exist once the CDI is created.

Reply:
I slightly prefer creating the handle since it gives each Crypto implementation some flexibility on creating it but there is no real advantage to it.

What do you think?

Reply:
I just mean creating the handle in `derive_exported_cdi`. So it would still be in the Crypto implementation, just a different place.

Reply:
Oh I understand now! 

That is much cleaner! Thanks! :D